### ISSUE TITLE
Confusing results when using step() - is step actually working?

### ISSUE THREAD

**Issue by TheEimer** [automl/arlbench#41]
I'm not sure if this is simply a very strange coincidence or if the step() function is not applying HP changes, but I'm getting the exact(!) same mean reward per AutoRL env step() call for the following variations of DQN on CartPole:
- switching learning rate from 0.003 to one of 5e-5, 0.001, 0.01, 0.05 at any one of {1,2, ... 9}/10 of training (yes, I ran all of these, only the seed causes values to change)
- switching exploration fraction from 0.1 to 0.01 or 0.7 after 50% of training
- switching inital_epsilon from 0.9 to one of 0.3, 0.7 or 0.01 after 50% of training

These seem like very strange results to me, I don't think it's realistic all of them should stay the same. My script looks like this:

```
from __future__ import annotations

import warnings

warnings.filterwarnings("ignore")
import logging
import sys
import traceback
from typing import TYPE_CHECKING

import numpy as np
import pandas as pd
import hydra
import jax
from arlbench.autorl import AutoRLEnv

if TYPE_CHECKING:
    from omegaconf import DictConfig


@hydra.main(version_base=None, config_path="configs", config_name="base")
def execute(cfg: DictConfig):
    """Helper function for nice logging and error handling."""
    logging.basicConfig(
        filename="job.log", format="%(asctime)s %(message)s", filemode="w"
    )
    logger = logging.getLogger()
    logger.setLevel(logging.INFO)

    if cfg.jax_enable_x64:
        logger.info("Enabling x64 support for JAX.")
        jax.config.update("jax_enable_x64", True)
    try:
        return run(cfg, logger)
    except Exception:
        traceback.print_exc(file=sys.stderr)
        raise


def run(cfg: DictConfig, logger: logging.Logger):
    """Console script for arlbench."""
    # Run until breakpoint
    env = AutoRLEnv(cfg.autorl)
    _ = env.reset()
    rewards = []
    grad_norms = []
    grad_vars = []
    hps = []
    for i in range(cfg.total_steps):
        if i == cfg.switching_point:
            logger.info(f"Switching hyperparameter at step {i}.")
            cfg.hp_config[cfg.target_hp] = cfg.target_hp_value
        obs, objectives, _, _, _ = env.step(cfg.hp_config)
        rewards.append(float(objectives["reward_mean"]))
        grad_norm, grad_var = obs["grad_info"]
        grad_norms.append(float(grad_norm))
        grad_vars.append(float(grad_var))
        hps.append(cfg.hp_config[cfg.target_hp])

    df = pd.DataFrame(
        {
            "step": list(range(cfg.total_steps)),
            "reward": rewards,
            "grad_norm": grad_norms,
            "grad_var": grad_vars,
            f"current_{cfg.target_hp}": hps,
            "switched": [
                i >= cfg.switching_point for i in range(cfg.total_steps)
            ],
            "switching_point": [cfg.switching_point] * cfg.total_steps,
            "seed": [cfg.autorl.seed] * cfg.total_steps,
            cfg.target_hp: [cfg.target_hp_value] * cfg.total_steps,
        }
    )

    df.to_csv("results.csv", index=False)


if __name__ == "__main__":
    sys.exit(execute())  # pragma: no cover
```

The config is the same as in the examples for the gradient schedule (total_num_timesteps is 1e4 to have 1/10 of runtime per step and gradient infos are returned). Any idea what's happening here? Is it my mistake or is the hp_config not applied?

**Comment by TheEimer** [comment]
It also doesn't seem to be env/algo specific, the same happens when running PPO on the brax fast config (same results for switching from lr 0.003 to either 0.09 or 0.001). Going through the code, I can't trivially find the issue apart from that possibly the hp_config is actually kept somewhere in the compiled bits (training state etc.) and thus isn't updated. Seems not the case at first glance, but maybe there are some jax shenanigans going on...
